modules/opensearch/src/OpenSearchTingObjectCollection.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * @file
+ * The OpenSearchTingCollection class.
+ */
+
+namespace OpenSearch;
+
+use Ting\TingObjectCollectionInterface;
+
+/**
+ * Class OpenSearchTingObjectCollection
+ *
+ * Wraps a TingClientObjectCollection to provide integration to Ding.
+ * @package OpenSearch
+ */
+class OpenSearchTingObjectCollection implements TingObjectCollectionInterface {
+
+  /**
+   * @var \OpenSearch\OpenSearchTingObject[]
+   */
+  protected $tingObjects = [];
+
+  /**
+   * OpenSearchTingCollection constructor.
+   *
+   * @param \TingClientObjectCollection $collection
+   *   Collection from the ting client.
+   */
+  public function __construct($collection) {
```

Comment:
How about adding a proper type hint for `TingClientObjectCollection` here as well?